sources/packages/api/src/services/application/application.service.ts

```diff
@@ -356,10 +356,19 @@ export class ApplicationService extends RecordDataModelService<Application> {
       })
       .getOne();
   }
-
+  /**
+   * Fetches application by applicationId and userId|studentId.
+   * When studentId is supplied, studentId is preferred as search criteria.
+   * When studentId is not supplied, then userId is used.
+   * @param applicationId
+   * @param userId
+   * @param studentId
+   * @returns
+   */
   async getApplicationByIdAndUser(
```

Comment:
Not a part of this PR, I found that this service is also used to retrieve student applications, 
![image](https://user-images.githubusercontent.com/77353155/140797405-2d14c50c-6731-4978-aced-713841207ab3.png),
 and I didn't find any logic or restriction not to show `overwritten` application on the API side, the frontend has the logic to show the form in read-only mode ( when a student tries to access the application by editing the application id from URL). We need to address this right @devs?

Reply:
Since this method is specific to returning the application for a user(student), I would assume that there is no need to return _overwritten_ application, as you mentioned @ann-aot.

Reply:
Feedback comment addressed.